flight_club/sessions/views.py

```diff
@@ -19,65 +19,253 @@
 from flight_club.models.models import Beer, Session
 from flight_club.auth.views import login_required
 from flight_club.sessions.fc_sessions import FCSession
+from werkzeug.exceptions import BadRequestKeyError
 
+import datetime
 import flight_club.models.db_func as db_func
 
 bp = Blueprint("sessions", __name__, url_prefix="/sessions")
 
 
+def session_input_validator(session_id, date, host):
+    """This function is for validation the inputs for the session portion
+    of add session. It will type check and value check.
+
+    Args:
+        session_id : the id of the session
+        date : the date of the session
+        host : the host of the
+
+    Returns:
+        boolean indicating if checks failed
+        error string, either error or None
+        session object or None if errors
+
+    """
+    # Session id checks
+    try:
+        session_id = int(session_id)
+    except:
+        return False, "Session_id needs to be a positive integer", None
+
+    if db_func.check_if_session_exists(session_id):
+        return False, "Session already exists", None
+
+    if session_id <= 0:
+        return False, "Session id should be a positive number, greater than 0", None
+
+    # Date checks
+    # Note date is passed as YYYY-MM-DD we want to store and display MM/DD/YYYY.
+    # Mostly because we are dumb Americans.
+    try:
+        date_object = datetime.datetime.strptime(date, "%Y-%m-%d")
+    except ValueError:
+        # This should never happen, since we use a date input selector.
+        # Maybe on day someone will attempt to hack us via the date field?
+        return False, "Date was not inputted in YYYY-MM-DD format", None
+
+    date = datetime.date.strftime(date_object, "%m/%d/%Y")
```

Comment:
There's a fair amount of stuff in a session.  We might want to consider making it an object with members that are ID, date, winner, etc.

Reply:
I'm a bit confused by the location of this comment.

This is really just form validation. This could get wrapped into some session class and then validation be performed on the member variables. However this is all occurring prior to insert to the database. The end of this function does make a `Session` object which is a SQLAlchemy model. I imagine some of the validation probably happens on creation of the `Session` object, but I wanted that to be a bit more clear. So this checking contents of the post request to make sure the user isn't passing anything invalid in, and providing an error back to the user if they did so. 

I could write some sort of validation wrapper around the Model class, but I don't know seemed a bit excessive.

Reply:
or I could just make the form validator it's own class...that creates the appropriate objects to shove in the database. That seems a bit cleaner. It will just take in the post request.

Reply:
Oh I didn't see the `Session` object creation at the bottom of the method.  I like the approach of validation being its own class or you can leave as is and make a follow up ticket

Reply:
I did it, because I realized I never would after the PR.